content/en/platform/corda/5.0-beta/developing/getting-started/cordapp-standard-development-environment/cordapp-chat.md

```diff
@@ -0,0 +1,391 @@
+
+* clientRequestId to uniquely identify the request
+
+* flowClassName which provides the fully qualified name of the flow to be triggered
+
+* requestData which provides the input arguments for the flow
+
+For example:
+
+```java
+   {
+    "clientRequestId": "create-1",
+    "flowClassName": "com.r3.developers.csdetemplate.utxoexample.workflows.CreateNewChatFlow",
+    "requestData": {
+        "chatName":"Chat with Bob",
+        "otherMember":"CN=Bob, OU=Test Dept, O=R3, L=London, C=GB",
+        "message": "Hello Bob"
+        }
+}
+ ```
+
+Swagger will also give you the curl command which can be used to run the request directly from the command line, for example:
+
+  ```java
+curl -X 'POST' \
+  'https://localhost:8888/api/v1/flow/17F49B05B2B5' \
+  -H 'accept: application/json' \
+  -H 'Content-Type: application/json' \
+  -d '{
+    "clientRequestId": "create-1",
+    "flowClassName": "com.r3.developers.csdetemplate.utxoexample.workflows.CreateNewChatFlow",
+    "requestData": {
+        "chatName":"Chat with Bob",
+        "otherMember":"CN=Bob, OU=Test Dept, O=R3, L=London, C=GB",
+        "message": "Hello Bob"
+        }
+}'
+```
+
+If the flow has been successfully started Swagger will show a “START REQEUSTED” response, for example:
+
+  ```java
+{
+  "holdingIdentityShortHash": "17F49B05B2B5",
+  "clientRequestId": "create-1",
+  "flowId": null,
+  "flowStatus": "START_REQUESTED",
+  "flowResult": null,
+  "flowError": null,
+  "timestamp": "2023-01-18T09:45:25.911889Z"
+}
+```
+
+If something has gone wrong you will get an error response.
+For polling for the result of a flow we will use the GET: /flow/{holdingidentityshorthash}/{clientrequestid} endpoint. This requires the short hash of the node the flow was run against and the clientRequestId specified when the flow was run.
+
+The curl version is:
+curl -X 'GET' \
+  'https://localhost:8888/api/v1/flow/17F49B05B2B5/create-1' \
+  -H 'accept: application/json'
+If the flow has run successfully this will return a “COMPLETED” Status together with the flowResult.
+
+  ```java
+{
+  "holdingIdentityShortHash": "17F49B05B2B5",
+  "clientRequestId": "create-1",
+  "flowId": "fa0dec91-4a94-476c-8b1b-82dc95a55134",
+  "flowStatus": "COMPLETED",
+  "flowResult": "SHA-256D:FACC6487F3478D69A543CE7061611C8792C0FBC1FCC4903742ABF82E0D8F1D14",
+  "flowError": null,
+  "timestamp": "2023-01-18T10:36:16.889777Z"
+}
+```
+
+{{< note >}}
+It can take up to a minute for Corda to Process the flow, this is likely a function of using the local Combined Worker version of Corda which runs all the cluster processes in one JVM with limited resources. If this were to be run in a more typical cloud deployment you would expect it to be much faster. Whilst corda is still processing the request you would get a “RUNNING” status returned. Keep polling the end point every 10 seconds or so until you get a result.
```

Comment:
```suggestion
It can take up to a minute for Corda to process the flow. This is likely a function of using the local Combined Worker version of Corda which runs all the cluster processes in one JVM with limited resources. If this were to be run in a more typical cloud deployment, you would expect it to be much faster. Whilst corda is still processing the request, Swagger returns a “RUNNING” status. Keep polling the end point every 10 seconds or so until you get a result.
```